File: frontend/pyside6/api/product_api.py
import requests

BASE_URL = "http://127.0.0.1:8000"

def get_products_by_category(category_id: int):
    response = requests.get(
        f"{BASE_URL}/products/category/{category_id}"
    )
    response.raise_for_status()
    return response.json()

def create_product(category_id: int, name: str, price: int, image_url: str = ""):
    response = requests.post(
        f"{BASE_URL}/products",
        json={
            "category_id": category_id,
            "name": name,
            "price": price,
            "image_url": image_url
        }
    )
    response.raise_for_status()
    return response.json()

File: frontend/react/src/api/ProductApi.ts
import axiosInstance from './axiosInstance';
import type { ProductCreate, ProductResponse } from '../types/product';

export const getProducts = async (categoryId: number): Promise<ProductResponse[]> => {
  const response = await axiosInstance.get(`/products/category/${categoryId}`);
  return response.data;
};

export const createProduct = async (data: ProductCreate): Promise<ProductResponse> => {
  const response = await axiosInstance.post('/products', data);
  return response.data;
};